### PR TITLE
DataFrame.drop_duplicates

### PR DESCRIPTION
**Pull request by wuisawesome**
## What do these changes do?

Implementation for DataFrame.drop_duplicates.
Tests for drop_duplicates under test_dataframe.test_drop_duplicates()

## Related issue number

N/A

- [✅] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [✅] passes `black --check modin/`

### CODE REVIEW

File: modin/pandas/dataframe.py [hunk 1]
```diff
@@ -1267,9 +1267,28 @@ def drop(
         return self._create_dataframe_from_compiler(new_query_compiler, inplace)
 
     def drop_duplicates(self, subset=None, keep="first", inplace=False):
-        return self._default_to_pandas(
-            pandas.DataFrame.drop_duplicates, subset=subset, keep=keep, inplace=inplace
-        )
+        """
```
Comment (devin-petersohn) [inline modin/pandas/dataframe.py]: Method level doc format: Single line description on the first line.

File: modin/pandas/dataframe.py [hunk 2]
```diff
@@ -1267,9 +1267,28 @@ def drop(
...
+                Return DataFrame with duplicate rows removed, optionally only
+                considering certain columns
+                Parameters
```
Comment (devin-petersohn) [inline modin/pandas/dataframe.py]:
```suggestion
            Args:
```

File: modin/pandas/dataframe.py [hunk 3]
```diff
@@ -1267,9 +1267,28 @@ def drop(
...
+                ----------
```
Comment (devin-petersohn) [inline modin/pandas/dataframe.py]: Delete

File: modin/pandas/dataframe.py [hunk 4]
```diff
@@ -1267,9 +1267,28 @@ def drop(
...
+                subset : column label or sequence of labels, optional
+                    Only consider certain columns for identifying duplicates, by
+                    default use all of the columns
+                keep : {'first', 'last', False}, default 'first'
+                    - ``first`` : Drop duplicates except for the first occurrence.
+                    - ``last`` : Drop duplicates except for the last occurrence.
+                    - False : Drop all duplicates.
+                inplace : boolean, default False
+                    Whether to drop duplicates in place or to return a copy
+                Returns
```
Comment (devin-petersohn) [inline modin/pandas/dataframe.py]:
```suggestion
            Returns:
```

File: modin/pandas/dataframe.py [hunk 5]
```diff
@@ -1267,9 +1267,28 @@ def drop(
...
+                -------
```
Comment (devin-petersohn) [inline modin/pandas/dataframe.py]: Delete

File: modin/pandas/test/test_dataframe.py [hunk 6]
```diff
@@ -1385,14 +1385,38 @@ def test_drop_api_equivalence():
...
-    modin_df = pd.DataFrame(data)
-    pandas_df = pandas.DataFrame(data)  # noqa F841
+def test_drop_duplicates():
+    frame_data = {
+        "A": list(range(3)) * 2,
+        "B": list(range(1, 4)) * 2,
+        "C": list(range(6)),
+    }
+    modin_df = pd.DataFrame(frame_data)
+    pandas_df = pandas.DataFrame(frame_data)  # noqa F841
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py]:
```suggestion
    pandas_df = pandas.DataFrame(frame_data)
```

File: modin/pandas/test/test_dataframe.py [hunk 7]
```diff
@@ -1385,14 +1385,38 @@ def test_drop_api_equivalence():
         modin_df.drop(axis=1)
 
 
-@pytest.mark.skip(reason="Defaulting to Pandas")
-@pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
-def test_drop_duplicates(data):
-    modin_df = pd.DataFrame(data)
-    pandas_df = pandas.DataFrame(data)  # noqa F841
+def test_drop_duplicates():
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py]: It might be good to do:

```python
@pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
def test_drop_duplicates(data):
```

It will require some small changes in the test code, but it will test on a wider variety of edge case DataFrames.

Comment (wuisawesome) [reply modin/pandas/test/test_dataframe.py]: That sounds like a good idea, though I think we need the full `test_data` dictionary since including the "float_data" column in searching for duplicates effectively guarantees that there will never a duplicate
